apps/microtvm/cmsisnn/Makefile

```diff
@@ -69,6 +69,18 @@ $(BUILD_DIR)/crt_backend_api.o: $(STANDALONE_CRT_PATH)/src/runtime/crt/common/cr
 	$(QUIET)mkdir -p $(@D)
 	$(QUIET)$(CC) -c $(PKG_CFLAGS) -o $@  $^
 
+$(BUILD_DIR)/crt_runtime_api.o: $(STANDALONE_CRT_PATH)/src/runtime/crt/common/crt_runtime_api.c
+	$(QUIET)mkdir -p $(@D)
+	$(QUIET)$(CC) -c $(PKG_CFLAGS) -o $@  $^
+
+$(BUILD_DIR)/func_registry.o: $(STANDALONE_CRT_PATH)/src/runtime/crt/common/func_registry.c
+	$(QUIET)mkdir -p $(@D)
+	$(QUIET)$(CC) -c $(PKG_CFLAGS) -o $@  $^
+
+$(BUILD_DIR)/ndarray.o: $(STANDALONE_CRT_PATH)/src/runtime/crt/common/ndarray.c
+	$(QUIET)mkdir -p $(@D)
+	$(QUIET)$(CC) -c $(PKG_CFLAGS) -o $@  $^
+
```

Comment:
I'm assuming these are dependencies for `crt_runtime_api.c`? We should raise an issue to remove that dependency if we can?

@areusch had some thoughts awhile back on this https://discuss.tvm.apache.org/t/pre-rfc-api-change-formalizing-c-backend-api/10380

Reply:
Yes, they are dependencies for `crt_runtime_api.c`  which defines `TVMAPISetLastError` and `TVMGetLastError`.

Instead of depending on `crt_runtime_api.c` we can generate the definitions of these functions as @ashutosh-arm  thought: https://github.com/apache/tvm/pull/13643#discussion_r1052140042, what do you think?

Reply:
If we don't emit the lines that set them, we shouldn't need the definitions at all?

Reply:
I am not sure I understood, can you elaborate?

In this pr, we save the error through `TVMAPISetLastError` and then we retrieve it through `TVMGetLastError`, so we need the definitions of these two functions.
In the current implementation the functions are defined in `crt_runtime_api.c`, the problem is that this file has other functions, which we do not use, that depend on functions defined in `func_registry.c` and `ndarray.c`.
Ashutosh proposed not to use the definitions in `crt_runtime_api.c`, but to just emit the definitions of these two functions in `aot.py`.

Reply:
There are two separate issues here:
1. The C Runtime doesn't make it easy to use only the bits you want so you suddenly need array handling if you want error handling - this is a future problem if we want to tackle it
2. If we emit calls to these functions we're breaking anyone who currently compiles their model with CMSIS-NN - we need to have a way for users to specify that they don't want the debugging behaviour and then we don't emit the calls to the `TVMAPISetLastError`/`TVMAPIGetLastError` apis.

Reply:
I've added an option `debug_last_error` to `CMSISNNCompilerConfigNode`.
The calls to `TVMAPISetLastError` and `TVMAPIGetLastError` are generated only if this option is true.
When the option is false we no longer need the dependencies: `crt_runtime_api.c`, `func_registry.c` and `ndarray.c`. We need them only when the option is true.
I've also added a test, that checks if a model compiles with both values of this option.

Reply:
Ping for review @Mousius. :)